examples/Custom Chain + Tool + Protect in Observe/Sample_Observe+Tool+Protect+NonLangchain.ipynb

```diff
@@ -0,0 +1,287 @@
+        "            },\n",
+        "        }\n",
+        "    }\n",
+        "]"
+      ],
+      "metadata": {
+        "id": "inqwSZsuwSXk"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Conversational Flow\n",
+        "inputs = [\n",
+        "{\"role\": \"system\", \"content\": \"You are a helpful customer support assistant. Use the supplied tools to assist the user.\"},\n",
+        "{\"role\": \"user\", \"content\": \"Hi, can you tell me the delivery date for my order?\"},\n",
+        "{\"role\": \"user\", \"content\": \"My order number is 12345.\"}\n",
+        "]"
+      ],
+      "metadata": {
+        "id": "OzzT0VwC_L_-"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "for i, prompt in enumerate(inputs):\n",
+        "    if prompt[\"role\"] == \"system\":\n",
+        "        wf = observe_logger.add_workflow(input={\"input\": prompt[\"content\"]}, output={\"output\": str(tools)}, name=\"CustomWorkflow\")\n",
+        "\n",
+        "    if prompt[\"role\"] == \"user\":\n",
+        "        current_messages = inputs[:i+1]\n",
+        "        # Conduct chat completion\n",
+        "        response = client.chat.completions.create(\n",
+        "          model=\"gpt-35-turbo\",\n",
+        "          messages=current_messages,\n",
+        "          tools=tools,\n",
+        "        )\n",
+        "        output_message = response.choices[0].message\n",
+        "\n",
+        "        # Check if it is an LLM or a Tool message\n",
+        "        if output_message.content is not None:\n",
+        "            wf.add_llm(\n",
+        "            input=prompt[\"content\"],\n",
+        "            output=output_message.content,\n",
+        "            model=\"gpt-35-turbo\",\n",
+        "            input_tokens=response.usage.prompt_tokens,\n",
+        "            output_tokens=response.usage.completion_tokens,\n",
+        "            total_tokens=response.usage.total_tokens,\n",
+        "            metadata={\"Node\": \"LLM\"}\n",
+        "            )\n",
+        "        elif output_message.tool_calls is not None:\n",
+        "            order_id = json.loads(response.choices[0].message.tool_calls[0].function.arguments)[\"order_id\"]\n",
+        "            tool_output = get_delivery_date(order_id)\n",
+        "            wf.add_tool(\n",
+        "            input=response.choices[0].message.tool_calls[0].function.arguments,\n",
+        "            output=f\"Your delivery date is {tool_output}.\",\n",
+        "            name=response.choices[0].message.tool_calls[0].function.name,\n",
+        "            metadata={\"Node\": \"Function\"}\n",
+        "            )\n",
+        "\n",
+        "        # On LLM steps, also run Protect\n",
+        "        if output_message.content is not None:\n",
```

Comment:
There's an `if output_message.content is not None` already. Combine the two? Or re-arrange the code?